src/ring_theory/non_zero_divisors.lean

```diff
@@ -73,4 +73,28 @@ lemma map_le_non_zero_divisors_of_injective {B : Type*} [integral_domain B]
 le_non_zero_divisors_of_domain (λ h, let ⟨x, hx, hx0⟩ := h in
   zero_ne_one (hM (hf (trans hx0 (f.map_zero.symm)) ▸ hx : 0 ∈ M) 1 (mul_zero 1)).symm)
 
+lemma prod_zero_iff_exists_zero {R : Type*} [comm_semiring R] [no_zero_divisors R] [nontrivial R]
+  {s : multiset R} : s.prod = 0 ↔ ∃ (r : R) (hr : r ∈ s), r = 0 :=
+begin
+split, swap,
+  { rintros ⟨r, hrs, hrz⟩,
+    apply multiset.prod_eq_zero,
+    rwa hrz at hrs },
+  refine multiset.induction _ (λ a s ih, _) s,
+  { rw multiset.prod_zero,
+    intro habs,
+    by_contra, from ne.elim (@one_ne_zero _ _ _) habs },
```

Comment:
Or just replace this block with:
```suggestion
  { intro habs,
    simpa using habs, },
```
(`simp` is your friend, and `simpa` is clever enough to run a "by contradiction" proof if it reduces the named hypothesis to `false`)

Reply:
I see, I'm always scared with using `simp` and `simpa`, but when they are closing tactics it is certainly wiser to use them. Thanks.